linker/shared/src/main/scala/org/scalajs/linker/backend/BasicLinkerBackend.scala

```diff
@@ -41,9 +41,11 @@ final class BasicLinkerBackend(config: LinkerBackendImpl.Config)
   private[this] var totalModules = 0
   private[this] val rewrittenModules = new AtomicInteger(0)
 
+  private[this] val fragmentIndex = new SourceMapWriter.Index
```

Comment:
Note: We accept leaking old sources / names in incremental runs here (similar to how NameGen leaks as well).

Reply:
Yes, because the `Emitter` is in fact single threaded. This is probably also something we should fix. I'll make `Index` thread-safe and see if it affects the benchmarks. If it doesn't, we should probably keep the thread safe version :).